### PR TITLE
More actions

### PR DESCRIPTION
**Pull request by toger5**
- [ ] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

File: crates/matrix-sdk/src/widget_api/messages/to_widget/actions.rs [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(tag = "action")]
 pub enum ToWidgetAction {
     #[serde(rename = "capabilities")]
     SendMeCapabilities,
     #[serde(rename = "notify_capabilities")]
-    CapabilitiesUpdated
+    CapabilitiesUpdated,
+    #[serde(rename = "notify_capabilities")]
+    OpenIdCredentials,
```
Comment (daniel-abramov) [inline crates/matrix-sdk/src/widget_api/messages/to_widget/actions.rs]: Just a question: is it expected to be empty?

File: crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs [hunk 2]
```diff
@@ -23,3 +27,32 @@ pub struct SupportedVersions {
...
+    matrix_server_name: Option<String>,
+    token_type: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct SendToDeviceRequest {
+    #[serde(rename = "type")]
+    message_type: String,
+    encrypted: bool,
+    content: HashMap<String, HashMap<String, serde_json::Value>>,
```
Comment (daniel-abramov) [inline crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs:48]: The usage of `serde_json::Value` gives some "might be wrong" vibes 🙂 Are we confident that we just pass this `Value` around without really parsing it?

**Review by daniel-abramov — COMMENTED** [review]
Added a couple of small comments.

Just another though: I think I would prefer us adding a message + its handler right away. That way for simple messages a PR can contain the structure definitions and the `handle()` part of it. For more complex message (open ID request) I think it would deserve its own PR since it probably would involve a bit more complicated handler part.

Otherwise we'll just end up having many unimplemented structures that may and will change once we start working on a handler 🙂

File: crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs (hunk 2 above)
Comment (toger5) [reply crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs:48]: the actual content of the to device request can be very generic: the content field itself has a user key and inside a user key the device keys. But the actual message can be anything (including custom json that a widget developer might want to send)

Comment (daniel-abramov) [reply crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs:48]: Thanks for the clarification! Should at least the keys be then somewhat specified? (some concrete type with a concrete name?)

Feel free to resolve upon reading :)

Comment (toger5) [reply crates/matrix-sdk/src/widget_api/messages/from_widget/actions.rs:48]: I think on this level not anymore. The event consists of type, and content. The fields of the contend depend on the type but the type can be any string. (there are of course specced types that define a content but it is also possible to introduce any new type in a widget with arbitrary content. -> at this stage one cannot tell anything about the structure of content)

**Comment by toger5** [comment]
closed in favor of #7

**Pull request closed without merging**